pyomo/contrib/doe/fim_doe.py

```diff
@@ -0,0 +1,2582 @@
+                else:
+                    measurement_accurate_time[j][no_t] = t_all[t_all.index(tt)]
+
+        # set ub and lb to parameters
+        for par in self.param_name:
+            component = getattr(mod, par)[0]
+            component.setlb(self.param_init[par])
+            component.setub(self.param_init[par])
+
+        # generate parameter name list and value dictionary with index
+        var_name = []
+        var_dict = {}
+        for name in self.param_name:
+            var_name.append(name+'[0]')
+            var_dict[name+'[0]'] = self.param_init[name]
+
+        # call k_aug get_dsdp function
+        time0_solve = time.time()
+        square_result = self.__solve_doe(mod, fix=True)
+        dsdp_re, col = get_dsdp(mod, var_name, var_dict, tee=self.tee_opt)
+        time1_solve = time.time()
+        time_solve = time1_solve - time0_solve
+
+        # analyze result
+        dsdp_array = dsdp_re.toarray().T
+        self.dsdp = dsdp_array
+        self.dsdp = col
+        # store dsdp returned
+        dsdp_extract = []
+        # get right lines from results
+        measurement_index = []
+        measurement_names = []
+        # produce the sensitivity for fixed variables
+        zero_sens = np.zeros(len(self.param_name))
+
+        # loop over measurement variables and their time points
+        for measurement_name in self.measure.model_measure_name:
+            # get right line number in kaug results
+            if self.discretize_model is not None:
+                # for DAE model, some variables are fixed
+                try:
+                    kaug_no = col.index(measurement_name)
+                    measurement_index.append(kaug_no)
+                    # get right line of dsdp
+                    dsdp_extract.append(dsdp_array[kaug_no])
+                except:
+                    self.logger.debug('The variable is fixed:', measurement_name)
+                    # for fixed variables, the sensitivity are a zero vector
+                    dsdp_extract.append(zero_sens)
+            else:
+                kaug_no = col.index(measurement_name)
+                measurement_index.append(kaug_no)
+                # get right line of dsdp
+                dsdp_extract.append(dsdp_array[kaug_no])
+
+        # Extract and calculate sensitivity if scaled by constants or parameters.
+        # Convert sensitivity to a dictionary
+        jac = {}
+        for par in self.param_name:
+            jac[par] = []
+
+        for d in range(len(dsdp_extract)):
+            for p, par in enumerate(self.param_name):
+                # if scaled by parameter value or constant value
+                if self.scale_nominal_param_value:
+                    jac[par].append(self.param_init[par]*dsdp_extract[d][p]*self.scale_constant_value)
+                else:
+                    jac[par].append(dsdp_extract[d][p]*self.scale_constant_value)
```

Comment:
This could be made simpler:
```python
jac = {}
for p, par in enumerate(self.param_name):
    if self.scale_nominal_param_value:
        scale = self.param_init[par]*self.scale_constant_value
    else:
        scale = self.scale_constant_value
    jac[par] = [d[p]*scale for d in dsdp_extract]
```